internal/utils_test.go

```diff
@@ -30,6 +30,82 @@ func assertReflectEqual(t *testing.T, a interface{}, b interface{}) {
 	}
 }
 
+func assertSelectorEqual(t *testing.T, a map[string]interface{}, b types.NamespaceSelector) {
+	t.Helper()
+
+	if !compareSelectors(a, b) {
+		t.Fatalf("%s != %s", a, b)
+	}
+}
+
+func compareSelectors(a map[string]interface{}, b types.NamespaceSelector) bool {
+	if includeA, ok := a["include"].([]interface{}); ok {
```

Comment:
I think it'd be easier to read if you just converted `a["include"]` to a slice of strings and then perform a reflect deep equal. Then you wouldn't have the concern of the slice lengths not being equal.

The same goes for `exclude`.